### PR TITLE
Revise swagger API documentation of TxMetadata

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-307 / #2073

### Overview

Updates swagger documentation according to recent API changes.

### Comments

[Rendered](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/input-output-hk/cardano-wallet/rvl/2073/tx-metadata-swagger/specifications/api/swagger.yaml)

### CODE REVIEW

**Review by KtorZ — APPROVED** [review]
Nice.

**Comment by KtorZ** [comment]
bors merge

File: specifications/api/swagger.yaml [hunk 1]
```diff
@@ -569,17 +569,33 @@ x-transactionMetadata: &transactionMetadata
...
-    are otherwise encoded as base16 will be decoded and stored as plain byte
-    strings on-chain (saving space and therefore, lowering fees).
+    4. Bytestrings are hex-encoded, with a maximum length of 64 bytes.
+
+    Metadata aren't stored as JSON on the Cardano blockchain but are
+    instead stored using a compact binary encoding (CBOR).
+
+    The binary encoding of metadata values supports three simple types:
+
+    * Integers in the range `-2^32` to `2^32 - 1`.
```
Comment (dcoutts) [inline specifications/api/swagger.yaml]: It's 64bit integers. -(2^64 - 1) .. 2^64 - 1

**Comment by KtorZ** [comment]
bors merge

**Comment by KtorZ** [comment]
bors retry

File: specifications/api/swagger.yaml (hunk 1 above)
Comment (rvl) [reply specifications/api/swagger.yaml]: Oops - thanks @dcoutts, sorry for stirring up confusion but...

... just one thing about this range - I know it's simply an artificial limit and nothing to do with machine representation of signed ints. But there are 2<sup>65</sup> - 1 possible values in this range, which might be unsettling for nerds.

Also there is a function `validateTxMetadataValue` in cardano-api which enforces the bounds [-2<sup>64</sup>, 2<sup>64</sup>]. So  2<sup>65</sup> + 1 possible values. I couldn't find the validation code in the consensus layer or specs.

**Pull request merged**